distributed/utils_test.py

```diff
@@ -651,98 +648,96 @@ def cluster(
         scheduler.daemon = True
         scheduler.start()
 
-        # Launch workers
-        workers = []
-        for i in range(nworkers):
-            q = mp_context.Queue()
-            fn = "_test_worker-%s" % uuid.uuid4()
-            kwargs = merge(
-                {
-                    "nthreads": 1,
-                    "local_directory": fn,
-                    "memory_limit": system.MEMORY_LIMIT,
-                },
-                worker_kwargs,
-            )
-            proc = mp_context.Process(
-                name="Dask cluster test: Worker",
-                target=_run_worker,
-                args=(q, scheduler_q, config),
-                kwargs=kwargs,
-            )
-            ws.add(proc)
-            workers.append({"proc": proc, "queue": q, "dir": fn})
+        with tempfile.TemporaryDirectory(prefix="_dask_test_worker") as tmpdirname:
+            # Launch workers
+            workers = []
+            for i in range(nworkers):
+                q = mp_context.Queue()
+                kwargs = merge(
+                    {
+                        "nthreads": 1,
+                        "local_directory": tmpdirname,
+                        "memory_limit": system.MEMORY_LIMIT,
+                    },
+                    worker_kwargs,
+                )
+                proc = mp_context.Process(
+                    name="Dask cluster test: Worker",
+                    target=_run_worker,
+                    args=(q, scheduler_q, config),
+                    kwargs=kwargs,
+                )
+                ws.add(proc)
```

Comment:
I see things being added to the `ws` weakset but then no further use? Is `ws` needed?

Reply:
there is a warning further down below to warn in case a process is still alive. Not great but I don't want to deal with this right now

Reply:
it's actually an assert, that's a bit better